skyportal/tests/api/test_candidates.py

```diff
@@ -1456,6 +1459,39 @@ def test_candidate_list_pagination(
     )
     assert status == 200
     assert data["data"]["candidates"][0]["id"] == obj_id1
+    assert "queryID" in data["data"]
+    query_id = data["data"]["queryID"]
+
+    status, data = api(
+        "GET",
+        "candidates",
+        params={"pageNumber": 1, "queryID": query_id},
+        token=view_only_token,
+    )
+    assert status == 200
+    assert data["data"]["queryID"] == query_id
+
+    # Wait until cache is expired
+    time.sleep(3)
+
+    # Run cron job that deletes old query cache files
+    root_project_dir = os.path.dirname(
+        os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    )
+    subprocess.call(
```

Comment:
This is not the way the system will be deployed in practice. To test this way, you need to install the cron job as usual, and wait for the timeout (not guaranteed to be 2 seconds, so probably wait 4) to see if the file disappeared.

I don't know what the planned timeout is for the current system, but it's worth noting that the cache items may be retained for up to 2x the number specified in the configuration.

Reply:
This is testing the script itself, just not the cron service, which has a minimum interval of one minute (longer than I think we want to wait in our tests)